cmd/juju/commands/debuglog.go

```diff
@@ -303,9 +379,89 @@ func (c *debugLogCommand) Run(ctx *cmd.Context) error {
 		c.params.NoTail = !isTerminal(ctx.Stdout)
 	}
 
+	// Get the controller addresses to connect to.
+	controllerAddr, err := c.getControllerAddresses()
+	if err != nil {
+		return err
+	}
+
+	// The default log buffer size is 1 for a single controller
+	// (stream log entries as they arrive).
+	bufferSize := 1
+	// If we are connecting to multiple controllers, adjust the buffer size so that
+	// we have a chance of ordering incoming records by timestamp.
+	// Replaying the logs will likely stream many initially so use a larger buffer size
+	// to account for controllers having potentially divergent log entry timestamps.
+	// This is best effort so it's ok if some log entries are printed out of order.
+	// Ideally we'd have a variable flush timeout depending on the rate of incoming messages
+	// but the buffered logger doesn't support that.
+	if len(controllerAddr) > 1 {
+		if c.params.Replay || c.params.NoTail {
+			bufferSize = 500
+		} else {
+			bufferSize = 5
+		}
+	}
+
+	buf := corelogger.NewBufferedLogger(logFunc(func(recs []corelogger.LogRecord) error {
+		for _, r := range recs {
+			_ = c.out.Write(ctx, &r)
+		}
+		return nil
+	}), bufferSize, time.Second, clock.WallClock)
+
+	// Start one log streamer per controller.
+	numStreams := 1
+	if n := len(controllerAddr); n > 0 {
```

Comment:
No, it's because the default debug-log behaviour is NOT to look up addresses and rely on the underlying base connection stuff in modelcmd (like any other juju command)